bcipy/gui/params_form.py

```diff
@@ -67,22 +65,37 @@ def createControls(self):
         parameters file."""
 
         self.controls = {}
-        for key, param in self.params.items():
-            if param.type == 'bool':
-                form_input = self.bool_input(param)
-            elif 'path' in param.type:
-                form_input = self.file_input(param)
-            elif isinstance(param.recommended_values, list):
-                form_input = self.selection_input(param)
-            # TODO: NumCtrl for numeric input types.
-            # from wx.lib.masked import NumCtrl
-            # elif param['type'] in ['float', 'int']:
+        # Retrieve and alphebatize list of unique parameter sections
+        param_sections = sorted(list(set([param[1].section for param in self.params.items()])))
+        # Sort parameters by section
+        key_dict = []
+        for each_section in param_sections:
+            # Get readable names for sections
+            name_dict = ast.literal_eval(self.params['section_names'].value)
+            new_key = name_dict[each_section] if each_section in name_dict else each_section
+            item_to_insert = [new_key, sorted(
+                [key for key, param in self.params.items() if param.section == each_section])]
+            if each_section == 'top_level_config':
+                # Move most important variables to top
+                key_dict.insert(0, item_to_insert)
```

Comment:
This whole section should be factored out into another method so we can test and confirm it's behavior. key_dict is an array, not a dict. In Python 3.6 and later the dict type maintains insertion order, however, a better data structure would be the collections.OrderedDict.

Reply:
Fixed! key_dict used to be a dictionary, and I forgot to change its name when I made it an array. It is now an OrderedDict.